core/src/Telemetry/XmsCliTelemInfoParser.cs

```diff
@@ -0,0 +1,94 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Identity.Core.Telemetry
+{
+    internal class XmsCliTelemInfoParser
+    {
+        private const string ExpectedCliTelemHeaderVersion = "1";
+        private const int ErrorCodeIndex = 1;
+        private const int SubErrorCodeIndex = 2;
+        private const int TokenAgeIndex = 3;
+        private const int SpeInfoIndex = 4;
+
+        internal XmsCliTelemInfo ParseXMsTelemHeader(string headerValue, RequestContext requestContext)
+        {
+            if (string.IsNullOrEmpty(headerValue))
+            {
+                return null;
+            }
+
+            string[] headerSegments = headerValue.Split(',');
+            if (headerSegments.Length == 0)
+            {
+                requestContext.Logger.Warning(
+                    string.Format(CultureInfo.InvariantCulture,
+                    TelemetryError.XmsCliTelemMalformed, headerValue));
+                return null;
+            }
+
+            string headerVersion = headerSegments[0];
+            XmsCliTelemInfo xMsTelemetryInfo = new XmsCliTelemInfo();
+            xMsTelemetryInfo.Version = headerVersion;
+
+            if (!string.Equals(headerVersion, ExpectedCliTelemHeaderVersion))
+            {
+                requestContext.Logger.Warning(
+                    string.Format(CultureInfo.InvariantCulture,
```

Comment:
This makes an indirect assumption that the string has one {0} parameter in it.  This can easily get broken and not noticed by the compiler.  Suggestion:  make a method for this that takes the one parameter and returns the formatted string?

Reply:
Fixed